core/domain/user_services_test.py

```diff
@@ -132,6 +137,52 @@ def test_get_user_id_from_username(self):
         self.assertIsNone(
             user_services.get_user_id_from_username('fakeUsername'))
 
+    def test_fetch_gravatar_success(self):
+        user_email = 'user@example.com'
+        gravatar_filepath = os.path.join(
+            'static', 'images', 'avatar', 'gravatar_example.png')
+        with open(gravatar_filepath, 'r') as f:
+            gravatar = f.read()
+        with self.urlfetch_mock(content=gravatar):
+            profile_picture = user_services.fetch_gravatar(user_email)
+            gravatar_data_url = utils.convert_png_to_data_url(gravatar_filepath)
+            self.assertEqual(profile_picture, gravatar_data_url)
+
+    def test_fetch_gravatar_failure_404(self):
+        user_email = 'user@example.com'
+        logging_error_mock = test_utils.CallCounter(lambda x: x)
+        urlfetch_counter = test_utils.CallCounter(urlfetch.fetch)
+        with nested(
+            self.urlfetch_mock(status_code=404),
+            self.swap(logging, 'error', logging_error_mock),
+            self.swap(urlfetch, 'fetch', urlfetch_counter)):
+            profile_picture = user_services.fetch_gravatar(user_email)
+            self.assertEqual(urlfetch_counter.times_called, 1)
+            self.assertEqual(logging_error_mock.times_called, 1)
```

Comment:
Hm, good question. How about:
```
    error_messages = []

    def mock_log_error(error_message):
        error_messages.append(error_message)
```

(all in the body of the test function) and swapping logging.error with mock_log_error instead? Then check the values in error_messages at the end.